spring-kafka/src/main/java/org/springframework/kafka/security/jaas/JaasLoginModuleConfiguration.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.kafka.security.jaas;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.security.auth.login.AppConfigurationEntry;
+
+import org.springframework.util.Assert;
+
+/**
+ * Contains properties for setting up an {@link AppConfigurationEntry} that can be used
+ * for the Kafka client.
+ *
+ * @author Marius Bogoevici
+ * @author Gary Russell
+ *
+ * @since 2.0
+ */
+public class JaasLoginModuleConfiguration {
+
+	private final Map<String, String> options = new HashMap<>();
+
+	private String loginModule = "com.sun.security.auth.module.Krb5LoginModule";
+
+	private AppConfigurationEntry.LoginModuleControlFlag controlFlag =
+			AppConfigurationEntry.LoginModuleControlFlag.REQUIRED;
+
+	public String getLoginModule() {
+		return this.loginModule;
+	}
+
+	public void setLoginModule(String loginModule) {
+		Assert.notNull(loginModule, "cannot be null");
+		this.loginModule = loginModule;
+	}
+
+	public String getControlFlag() {
+		return this.controlFlag.toString();
+	}
+
+	public void setControlFlag(String controlFlag) {
+		Assert.notNull(controlFlag, "cannot be null");
+		if (AppConfigurationEntry.LoginModuleControlFlag.OPTIONAL.equals(controlFlag)) {
```

Comment:
How about to switch to `switch`?
http://docs.oracle.com/javase/7/docs/technotes/guides/language/strings-switch.html